sqlsmith:  virtual column cannot reference a foreign key

sql/randgen.TestPopulateTableWithRandData [failed](https://mesolite.cluster.engflow.com/invocations/default/13c508d3-7f08-432e-9e2b-9fe3d3e5b23b?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL3JhbmRnZW46cmFuZGdlbl90ZXN0) on master @ [c7ef50846c992a99d5cd60bff8373c6df1c195a5](https://github.com/cockroachdb/cockroach/commits/c7ef50846c992a99d5cd60bff8373c6df1c195a5):


```
=== RUN   TestPopulateTableWithRandData
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestPopulateTableWithRandData31914337
    test_log_scope.go:81: use -show-logs to present logs inline
    schema_test.go:63: error executing query="CREATE TABLE \"%'table_1\" (\"col\\\\x581_0\" TIMETZ[] NOT NULL, col1_1 STRING COLLATE en_US NOT NULL, \"col%v1_2\" BIT(32), \"col 1_3\" BOX2D NOT NULL, col1_4 INT8 NOT NULL, \" col1_5\" INET NOT NULL, \"co%vl1_6\" BOOL, col1_7 TIMESTAMP NOT NULL, \"co\\gl1_8\" STRING NOT NULL AS (lower(CAST(col1_1 AS STRING))) STORED, col1_9 STRING NOT NULL AS (lower(CAST(\"col 1_3\" AS STRING))) VIRTUAL, \"c\"\"ol1_10\" STRING NOT NULL AS (CASE WHEN \"col\\\\x581_0\" IS NULL THEN e'g\\x13s:g/Rz``':::STRING ELSE e'``ol\"\\'\\n]':::STRING END) VIRTUAL, PRIMARY KEY (\"c\"\"ol1_10\" ASC, \"co\\gl1_8\" DESC), UNIQUE (\"col%v1_2\", \"c\"\"ol1_10\" DESC, \"col\\\\x581_0\" DESC), UNIQUE (\"co\\gl1_8\"));\nCREATE TABLE table_2 (\"c%pol2😖_0\" TIMETZ[] NULL, \"c̭'ol%p2_1\" INT4 NULL, col2_2 INTERVAL, col2_3 STRING NOT NULL, col2_4 OID NULL, col2__5 GEOGRAPHY NOT NULL, \"😉col2%v_6\" CHAR NULL, col2_7 INT8 NOT NULL, \"😮c%qol2_8\" FLOAT4 NOT NULL, \"co\"\"\\\\u19E7l2__9\" OID[] NOT NULL, col2😀_10 BOOL NOT NULL, \"co l2_11\" FLOAT8 NOT NULL, col2_12 \"char\" NOT NULL, col2_13 INT4 NOT NULL, \"cOl%v\f2_14\" FLOAT8 NOT NULL, col2_15 GEOGRAPHY NOT NULL, col2_16 NAME NOT NULL, col2_17 INT2 NOT NULL, PRIMARY KEY (col2_12), INVERTED INDEX (\"cOl%v\f2_14\" DESC, \"co l2_11\" ASC, col2_4 ASC, (CASE WHEN \"c%pol2😖_0\" IS NULL THEN e'\\x07]&GR\\x1ax':::STRING ELSE e'c\\x17F\\x0fI':::STRING END) ASC, ((col2_7 + col2_13) + col2_17) DESC, (CASE WHEN col2_2 IS NULL THEN 'l1':::STRING ELSE e'\\x01Q\\x02Y$D0':::STRING END), col2_7 DESC, col2_16 ASC, col2_3 DESC, col2__5 ASC) WHERE (((((((((((NOT table_2.col2😀_10) AND (table_2.col2_17 < (-32768):::INT8)) OR (table_2.\"😉col2%v_6\" < e'\\U00002603':::STRING)) AND (table_2.col2_7 <= 32767:::INT8)) OR (table_2.\"cOl%v\f2_14\" >= 1.0:::FLOAT8)) AND (table_2.col2_12 >= 'X':::STRING)) AND (table_2.\"😮c%qol2_8\" >= 3.4028234663852886e+38:::FLOAT8)) AND (table_2.\"c̭'ol%p2_1\" > 127:::INT8)) AND (table_2.col2_3 >= 'X':::STRING)) OR (table_2.col2_13 <= (-1):::INT8)) AND (table_2.\"co l2_11\" != (-1.0):::FLOAT8)) AND (table_2.col2_16 > e'\\'':::STRING:::NAME), INDEX (\"co\"\"\\\\u19E7l2__9\") STORING (col2_2, col2_4, col2__5, col2_7, \"😮c%qol2_8\", col2😀_10, col2_13, \"cOl%v\f2_14\") WHERE ((table_2.\"cOl%v\f2_14\" >= 1.401298464324817e-45:::FLOAT8) OR (table_2.\"c̭'ol%p2_1\" != 1:::INT8)) OR (table_2.col2_13 >= 0:::INT8), INDEX (\"c%pol2😖_0\" ASC), INVERTED INDEX (col2_15) WHERE (((((((((((table_2.\"cOl%v\f2_14\" >= 5e-324:::FLOAT8) OR table_2.col2😀_10) AND (table_2.\"co l2_11\" < 0.0:::FLOAT8)) OR (table_2.col2_13 >= (-2147483648):::INT8)) OR (table_2.col2_7 >= (-2147483648):::INT8)) AND (table_2.col2_16 = e'\\U00002603':::STRING:::NAME)) OR (table_2.\"😮c%qol2_8\" != '+Inf':::FLOAT8)) OR (table_2.\"😉col2%v_6\" != '\"':::STRING)) OR (table_2.\"c̭'ol%p2_1\" <= (-128):::INT8)) AND (table_2.col2_12 <= 'X':::STRING)) AND (table_2.col2_17 != (-1):::INT8)) AND (table_2.col2_3 < '':::STRING), INDEX (col2_16, \"cOl%v\f2_14\" ASC) PARTITION BY LIST (col2_16, \"cOl%v\f2_14\") (PARTITION \"table_2_\fp̛ar̽t_0\" VALUES IN ((e'\\x01X':::STRING:::NAME, (-0.08812231345628821):::FLOAT8), (e'I\\x17Z\\x15':::STRING:::NAME, 1.7976931348623157e+308:::FLOAT8), ('':::STRING:::NAME, (-1.0765701703139183):::FLOAT8), (e'7<2_hr\\x0fM':::STRING:::NAME, 0.6511553577012557:::FLOAT8), (e'gr^\\x14\\x1d':::STRING:::NAME, 0.38492207714317983:::FLOAT8), (e'\\x1b\\x10Py':::STRING:::NAME, 0.7066733933535441:::FLOAT8), ('pM':::STRING:::NAME, (-0.7662400858161311):::FLOAT8), ('1.y.':::STRING:::NAME, 0.753108501127421:::FLOAT8), (e':\\x18\\x04\\x02x\\x19y':::STRING:::NAME, 2.624621841175898:::FLOAT8), (e'F\\x16*':::STRING:::NAME, 1.1398599538589957:::FLOAT8)), PARTITION \"DEFAULT\" VALUES IN ((DEFAULT, DEFAULT))), INDEX (col2😀_10 DESC, \"c%pol2😖_0\" DESC, \"co\"\"\\\\u19E7l2__9\" ASC, \"cOl%v\f2_14\" ASC, \"😉col2%v_6\") STORING (col2_2, col2_3, col2_4, col2_7, \"😮c%qol2_8\", col2_15, col2_17), INDEX (\"co\"\"\\\\u19E7l2__9\" DESC) STORING (col2_4, col2__5, \"😉col2%v_6\", \"😮c%qol2_8\", col2😀_10, \"co l2_11\", col2_15, col2_16, col2_17) WHERE (((((((((((table_2.col2_13 != (-128):::INT8) AND (table_2.\"co l2_11\" > (-0.0):::FLOAT8)) OR (table_2.\"cOl%v\f2_14\" > 0.0:::FLOAT8)) AND (table_2.col2_7 > 0:::INT8)) OR (table_2.\"😮c%qol2_8\" <= 1.0:::FLOAT8)) AND (table_2.col2_17 <= (-1):::INT8)) OR table_2.col2😀_10) OR (table_2.col2_16 > 'X':::STRING:::NAME)) OR (table_2.\"😉col2%v_6\" = 'X':::STRING)) AND (table_2.col2_12 = e'\\U00002603':::STRING)) AND (table_2.\"c̭'ol%p2_1\" <= 127:::INT8)) AND (table_2.col2_3 <= e'\\x00':::STRING), INDEX (col2_13 DESC) WHERE table_2.col2_16 != e'\\'':::STRING:::NAME, UNIQUE (\"co\"\"\\\\u19E7l2__9\" ASC, col2_13) STORING (\"c%pol2😖_0\", \"c̭'ol%p2_1\", col2_4, col2__5, col2😀_10, col2_16, col2_17));\nCREATE TABLE table_3 (col3_0 DECIMAL NOT NULL, col3ͫ_1 CHAR NOT NULL, \"c\\rol3%p_2\" STRING NOT NULL AS (lower(col3ͫ_1)) VIRTUAL, PRIMARY KEY (col3ͫ_1 DESC, \"c\\rol3%p_2\" DESC, col3_0 ASC), INDEX (col3_0 ASC) WHERE table_3.col3_0 >= (-1):::DECIMAL, UNIQUE (\"c\\rol3%p_2\") WHERE ((table_3.col3ͫ_1 < e'\\x00':::STRING) OR (table_3.col3_0 < 1.0:::DECIMAL)) AND (table_3.\"c\\rol3%p_2\" <= '':::STRING));\nCREATE TABLE tabl🙀e_4 (col4_0 INTERVAL NOT NULL, col4_1 VARCHAR NOT NULL, \"😄col͑\f4_2\" REFCURSOR NULL, \"cOl4_3\" REGROLE, \"col 4_4\" JSONB NOT NULL, col4_5 OID, \"%pco🙃l4_6\" STRING NOT NULL AS (lower(col4_1)) STORED, col4_7 STRING AS (lower(CAST(col4_5 AS STRING))) VIRTUAL, \"col4\t_8\" STRING NOT NULL AS (lower(CAST(\"col 4_4\" AS STRING))) STORED, \"col4%_9\" STRING NOT NULL AS (lower(CAST(\"col 4_4\" AS STRING))) VIRTUAL, INVERTED INDEX (col4_7, \"col4\t_8\" gin_trgm_ops) WHERE ((((tabl🙀e_4.\"col4\t_8\" != '':::STRING) OR (tabl🙀e_4.col4_7 > 'X':::STRING)) AND (tabl🙀e_4.col4_1 < '\"':::STRING)) OR (tabl🙀e_4.\"%pco🙃l4_6\" <= e'\\'':::STRING)) OR (tabl🙀e_4.\"col4%_9\" >= e'\\U00002603':::STRING) NOT VISIBLE, UNIQUE (lower(CAST(\"😄col͑\f4_2\" AS STRING)) ASC) WHERE ((((tabl🙀e_4.col4_7 = '\"':::STRING) OR (tabl🙀e_4.col4_1 = e'\\U00002603':::STRING)) AND (tabl🙀e_4.\"col4%_9\" <= '\"':::STRING)) OR (tabl🙀e_4.\"%pco🙃l4_6\" > '\"':::STRING)) AND (tabl🙀e_4.\"col4\t_8\" > '':::STRING), UNIQUE (\"col 4_4\" DESC, col4_5 DESC) STORING (col4_1, \"😄col͑\f4_2\", \"cOl4_3\", \"%pco🙃l4_6\"), FAMILY (\"%pco🙃l4_6\", \"😄col͑\f4_2\"), FAMILY (col4_5, \"col4\t_8\", \"cOl4_3\"), FAMILY (col4_0), FAMILY (col4_1), FAMILY (\"col 4_4\"));\nCREATE TABLE ta͡ble_5 (col5_0 TSQUERY, \"co-l5_1\" TIMETZ NOT NULL, col5_2 TSVECTOR NOT NULL, \"\vcol\\\\u80CB5_3\" STRING NOT NULL AS (lower(CAST(col5_2 AS STRING))) VIRTUAL, col5_4 STRING NOT NULL AS (lower(CAST(\"co-l5_1\" AS STRING))) STORED, PRIMARY KEY (\"\vcol\\\\u80CB5_3\" ASC), INVERTED INDEX (col5_2 ASC));\nCREATE TABLE table_6 (col6_0 OID NULL, col6_1 PG_LSN NOT NULL, \"col6\v_2\" DECIMAL NOT NULL, \"%qcol6_3\" FLOAT8 NOT NULL, \"c̰\fol6_4\" UUID NOT NULL, \"col6/_5\" TIMETZ NOT NULL, col6_6 DECIMAL NULL, \"c😫o%vL6͂_7\" REGTYPE, col6_8 TIME NOT NULL, \"co%pl6_9\" DECIMAL NULL AS (\"col6\v_2\" + col6_6) VIRTUAL, \"\\ncol6_10\" STRING NOT NULL AS (lower(CAST(\"c̰\fol6_4\" AS STRING))) VIRTUAL, \"Col6_11\" DECIMAL NULL AS (\"col6\v_2\" + col6_6) STORED, \"coL6_12\" DECIMAL NOT NULL AS (\"col6\v_2\" + col6_6) STORED, col6_13 DECIMAL NOT NULL AS (\"col6\v_2\" + 8.847710389452341093E+22:::DECIMAL) STORED, col6_14 STRING NOT NULL AS (lower(CAST(\"col6/_5\" AS STRING))) STORED, PRIMARY KEY (\"coL6_12\", \"c̰\fol6_4\" ASC, \"col6\v_2\" ASC), INVERTED INDEX (col6_14 gin_trgm_ops) WHERE (((((table_6.\"%qcol6_3\" != '-Inf':::FLOAT8) AND (table_6.\"Col6_11\" < 'Infinity':::DECIMAL)) OR (table_6.\"co%pl6_9\" != 'NaN':::DECIMAL)) OR (table_6.\"coL6_12\" < (-1.0):::DECIMAL)) AND (table_6.col6_6 > (-1.0):::DECIMAL)) AND (table_6.\"\\ncol6_10\" > e'\\'':::STRING), INDEX ((\"col6\v_2\" + col6_6) ASC, \"co%pl6_9\") WHERE ((((table_6.\"coL6_12\" >= 1.0:::DECIMAL) OR (table_6.\"Col6_11\" != 1.0:::DECIMAL)) AND (table_6.\"\\ncol6_10\" < '':::STRING)) OR (table_6.col6_6 >= 1.0:::DECIMAL)) AND (table_6.\"col6/_5\" < '00:00:00+15:59':::TIMETZ), UNIQUE (col6_1, \"coL6_12\" ASC) STORING (col6_0, col6_8) WHERE ((((((((table_6.\"col6\v_2\" = 'Infinity':::DECIMAL) AND (table_6.\"co%pl6_9\" > 0:::DECIMAL)) OR (table_6.col6_8 != '24:00:00':::TIME)) AND (table_6.\"%qcol6_3\" <= 5e-324:::FLOAT8)) OR (table_6.col6_6 = 'Infinity':::DECIMAL)) AND (table_6.\"coL6_12\" > (-1.0):::DECIMAL)) AND (table_6.\"col6/_5\" != '00:00:00+15:59':::TIMETZ)) OR (table_6.col6_13 < 1.0:::DECIMAL)) OR (table_6.\"\\ncol6_10\" != 'X':::STRING), UNIQUE (lower(CAST(col6_8 AS STRING)), \"c😫o%vL6͂_7\" DESC), INDEX (col6_1 DESC) WHERE table_6.\"col6\v_2\" > 0:::DECIMAL NOT VISIBLE, INDEX (col6_14 DESC, \"col6\v_2\" ASC, col6_6), FAMILY (col6_1), FAMILY (col6_0), FAMILY (\"c̰\fol6_4\"), FAMILY (\"Col6_11\", col6_14), FAMILY (\"coL6_12\", col6_8), FAMILY (\"col6/_5\"), FAMILY (col6_13), FAMILY (col6_6, \"%qcol6_3\"), FAMILY (\"col6\v_2\"), FAMILY (\"c😫o%vL6͂_7\"), UNIQUE (col6_14, \"col6\v_2\"));\nCREATE TABLE table_7 (col7_0 DATE NOT NULL, col7_1 FLOAT8 NOT NULL, co😁l7_2 TIMESTAMP NOT NULL, col7_3 BYTES[] NOT NULL, col7_4 INTERVAL NOT NULL, \"c ol7%v_5\" BIT(34) NOT NULL, \"'col}7_6\" TSQUERY NOT NULL, \"co\"\"l/%557_7\" OID NULL, \"col\"\"7_8\" BIT(23), col7_9 DECIMAL[] NOT NULL, \"c%pol7%p_10\" BIT(26) NOT NULL, col7_11 INT4, \"%qcol7_12\" OID, \"c\tol😤7\\\\u5565_13\" CHAR NOT NULL, \"c&ol\t7_14\" OID NULL, col7_15 PG_LSN NOT NULL, \"col7/_16\" STRING NOT NULL AS (lower(CAST(\"c%pol7%p_10\" AS STRING))) VIRTUAL, col7_17 STRING NOT NULL AS (CASE WHEN col7_3 IS NULL THEN e'\\x1faW\\x1f\\x04':::STRING ELSE 'KD05)':::STRING END) VIRTUAL, PRIMARY KEY (\"c ol7%v_5\"), UNIQUE (co😁l7_2, col7_3, \"col\"\"7_8\") WHERE table_7.col7_11 >= (-128):::INT8, INDEX (col7_3 ASC, col7_15 DESC, \"col\"\"7_8\" ASC) STORING (col7_0, co😁l7_2, \"co\"\"l/%557_7\", col7_11, \"%qcol7_12\", \"c\tol😤7\\\\u5565_13\", \"c&ol\t7_14\"), INDEX (col7_0 ASC, lower(CAST(\"co\"\"l/%557_7\" AS STRING)) DESC, \"%qcol7_12\") STORING (col7_1, col7_4, \"'col}7_6\", \"co\"\"l/%557_7\", col7_11), UNIQUE (col7_4 DESC) WHERE ((((((table_7.col7_1 <= 1.7976931348623157e+308:::FLOAT8) AND (table_7.co😁l7_2 = '4714-11-24 00:00:00 BC':::TIMESTAMP)) AND (table_7.\"col7/_16\" <= e'\\x00':::STRING)) AND (table_7.\"c\tol😤7\\\\u5565_13\" = '\"':::STRING)) OR (table_7.col7_0 > '1970-01-01':::DATE)) AND (table_7.col7_17 != e'\\U00002603':::STRING)) AND (table_7.col7_11 > 1:::INT8), UNIQUE (col7_9 DESC, col7_3 ASC, \"c ol7%v_5\") STORING (co😁l7_2, \"co\"\"l/%557_7\", \"c%pol7%p_10\", col7_11, \"c\tol😤7\\\\u5565_13\", \"c&ol\t7_14\", col7_15) WHERE (((table_7.\"col7/_16\" <= 'X':::STRING) AND (table_7.col7_11 != (-32768):::INT8)) OR (table_7.col7_0 >= '-infinity':::DATE)) AND (table_7.col7_17 >= e'\\x00':::STRING), UNIQUE (col7_1 ASC, \"c\tol😤7\\\\u5565_13\"), INVERTED INDEX (\"%qcol7_12\" ASC, col7_17 gin_trgm_ops ASC) WHERE ((((table_7.\"c\tol😤7\\\\u5565_13\" <= '':::STRING) OR (table_7.col7_17 < 'X':::STRING)) OR (table_7.col7_1 >= '-Inf':::FLOAT8)) AND (table_7.col7_11 <= 32767:::INT8)) OR (table_7.\"col7/_16\" <= 'X':::STRING));\nCREATE TABLE table_8 (\"c%qol8_0\" STRING NOT NULL, col8_1 VARCHAR NOT NULL, \" co\\\\U000F0463l8_2\" INTERVAL NOT NULL, \" col8_3\" JSONB NOT NULL, \"col\\\\u50A3\t8_4\" STRING NOT NULL, col8_5 \"char\" NOT NULL, col8_6 INTERVAL NOT NULL, col8_7 REGCLASS NULL, \"col.%v8_8\" STRING NOT NULL AS (lower(col8_5)) VIRTUAL, \"c ol8_9\" STRING NOT NULL AS (lower(col8_1)) VIRTUAL, \"col8\\\\u36E0_10\" STRING NOT NULL AS (lower(col8_5)) STORED, col8_11 STRING NOT NULL AS (CASE WHEN col8_6 IS NULL THEN e' \\x1e4X?V':::STRING ELSE 'X':::STRING END) VIRTUAL, UNIQUE (lower(CAST(\" col8_3\" AS STRING)) ASC, \"col.%v8_8\" DESC, col8_7 ASC) STORING (\"c%qol8_0\") WHERE (((((table_8.col8_5 >= e'\\x00':::STRING) OR (table_8.\"col.%v8_8\" >= e'\\'':::STRING)) AND (table_8.\"col\\\\u50A3\t8_4\" != e'\\'':::STRING)) AND (table_8.\"col8\\\\u36E0_10\" < e'\\'':::STRING)) AND (table_8.col8_1 = '\"':::STRING)) AND (table_8.\"c%qol8_0\" != '\"':::STRING), INDEX (\"col.%v8_8\" DESC) STORING (\"c%qol8_0\", \" co\\\\U000F0463l8_2\", col8_7, \"col8\\\\u36E0_10\"), INDEX (col8_6) WHERE ((((((table_8.col8_1 > '':::STRING) OR (table_8.\"col8\\\\u36E0_10\" = e'\\U00002603':::STRING)) OR (table_8.\"col.%v8_8\" <= e'\\x00':::STRING)) AND (table_8.\"c ol8_9\" >= '\"':::STRING)) OR (table_8.\"col\\\\u50A3\t8_4\" = e'\\U00002603':::STRING)) AND (table_8.col8_5 != '':::STRING)) OR (table_8.col8_11 != '\"':::STRING), INDEX (\"c%qol8_0\", \" co\\\\U000F0463l8_2\" ASC) STORING (col8_1, \"col\\\\u50A3\t8_4\", col8_6, col8_7), UNIQUE (col8_6) WHERE (((((((table_8.\"col\\\\u50A3\t8_4\" != '':::STRING) OR (table_8.\"c ol8_9\" > e'\\'':::STRING)) OR (table_8.col8_5 != e'\\x00':::STRING)) OR (table_8.\"c%qol8_0\" < e'\\'':::STRING)) OR (table_8.\"col8\\\\u36E0_10\" >= e'\\x00':::STRING)) AND (table_8.col8_1 < 'X':::STRING)) AND (table_8.\"col.%v8_8\" != e'\\U00002603':::STRING)) AND (table_8.col8_11 <= e'\\U00002603':::STRING));\nCREATE TABLE table_9 (\"'col9_0\" JSONB, col9_1 TIMETZ NOT NULL, col9_2 DATE, \"col%q9_3\" STRING NOT NULL, col9_4 UUID NOT NULL, c̒ol9_5 INET, col9_6 PG_LSN NOT NULL, c😮ol9_7 DATE NOT NULL, col9_8 REFCURSOR NOT NULL, \"co\"\"%pl9_9\" REGTYPE, c̔ol9_10 STRING NOT NULL AS (lower(CAST(col9_8 AS STRING))) STORED, col9_11 STRING NOT NULL AS (lower(CAST(col9_8 AS STRING))) STORED, \"c'o\\fl9_12\" STRING NOT NULL AS (lower(CAST(col9_4 AS STRING))) VIRTUAL, col9_13 STRING NOT NULL AS (lower(CAST(col9_6 AS STRING))) STORED, col9_14 STRING NOT NULL AS (lower(CAST(col9_4 AS STRING))) VIRTUAL, col9_15 STRING NOT NULL AS (lower(\"col%q9_3\")) STORED, col9_16 STRING NOT NULL AS (lower(CAST(col9_6 AS STRING))) STORED, \"\vcol9_17\" STRING NOT NULL AS (lower(CAST(col9_6 AS STRING))) VIRTUAL, \"Col9_18\" STRING AS (CASE WHEN \"co\"\"%pl9_9\" IS NULL THEN NULL ELSE 'xqO':::STRING END) STORED, PRIMARY KEY (\"\vcol9_17\" ASC), INVERTED INDEX (c̔ol9_10 gin_trgm_ops) NOT VISIBLE, UNIQUE (col9_6) STORING (\"'col9_0\", c̒ol9_5, c😮ol9_7, col9_8) WHERE ((((((((table_9.\"c'o\\fl9_12\" <= e'\\x00':::STRING) OR (table_9.\"\vcol9_17\" != '':::STRING)) OR (table_9.col9_1 <= '24:00:00-15:59':::TIMETZ)) AND (table_9.col9_15 != 'X':::STRING)) OR (table_9.\"col%q9_3\" != e'\\x00':::STRING)) AND (table_9.c😮ol9_7 < '1970-01-01':::DATE)) AND (table_9.c̔ol9_10 <= e'\\x00':::STRING)) OR (table_9.col9_13 >= e'\\U00002603':::STRING)) AND (table_9.col9_2 != '1970-01-01':::DATE), FAMILY (col9_11, col9_13), FAMILY (col9_1, c̔ol9_10, col9_8), FAMILY (col9_16, \"Col9_18\"), FAMILY (c̒ol9_5), FAMILY (\"'col9_0\"), FAMILY (col9_2), FAMILY (col9_6), FAMILY (\"co\"\"%pl9_9\"), FAMILY (col9_4, \"col%q9_3\", c😮ol9_7), FAMILY (col9_15));\nCREATE TABLE table_10 (\"c\"\"ol10_0\" FLOAT4 NOT NULL, col10_1 JSONB NOT NULL, col10_2 INT8 NOT NULL, col10_3 STRING NOT NULL AS (lower(CAST(col10_1 AS STRING))) STORED, \"col 1%ab0_4\" INT8 NOT NULL AS (col10_2 + (-1388298835527789019):::INT8) STORED, PRIMARY KEY (\"c\"\"ol10_0\" DESC), UNIQUE (\"c\"\"ol10_0\" DESC, \"col 1%ab0_4\"), FAMILY (col10_1, \"c\"\"ol10_0\"), FAMILY (col10_2), FAMILY (col10_3, \"col 1%ab0_4\"));\nALTER TABLE table_8 ADD FOREIGN KEY (col8_11) REFERENCES \"%'table_1\" (\"co\\gl1_8\") ON DELETE RESTRICT;\nALTER TABLE table_3 ADD FOREIGN KEY (col3ͫ_1, col3_0) REFERENCES table_6 (col6_14, \"col6\v_2\") ON DELETE RESTRICT;\n" args=[]: pq: unimplemented: virtual column "col8_11" cannot reference a foreign key
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestPopulateTableWithRandData31914337
--- FAIL: TestPopulateTableWithRandData (0.77s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPopulateTableWithRandData.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39276

sql/randgen.TestPopulateTableWithRandData [failed](https://tanzanite.cluster.engflow.com/invocations/default/6287aeea-7c39-42e9-a113-00661c807d4c?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL3JhbmRnZW46cmFuZGdlbl90ZXN0) on master @ [d0582983521533ffc15a8b8525a188d80da2a0b7](https://github.com/cockroachdb/cockroach/commits/d0582983521533ffc15a8b8525a188d80da2a0b7):


```
=== RUN   TestPopulateTableWithRandData
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestPopulateTableWithRandData861088045
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    schema_test.go:63: error executing query="CREATE TABLE ta͙ble_1 (\"\\\\xa7c ol1_0\" PG_LSN, \"%vcol1_1\" GEOMETRY NULL, col1_2 DECIMAL NOT NULL, col1_3 REGROLE, col1_4 PG_LSN NOT NULL, co̞😥l1_5 TSQUERY, col1_6 TIMESTAMP NOT NULL, col1_7 REGTYPE NULL, \"\\\\x49col1_8\" INT2 NULL, col1_9 TIMETZ NOT NULL, col1_10 INT8 NULL AS (\"\\\\x49col1_8\" + 16281:::INT8) VIRTUAL, col1_11 STRING NOT NULL AS (lower(CAST(col1_9 AS STRING))) VIRTUAL, \"col 1_12\" DECIMAL NOT NULL AS (col1_2 + 59068031.10632513745:::DECIMAL) VIRTUAL, col1_13 STRING AS (CASE WHEN col1_3 IS NULL THEN e'\\x05t\\x06':::STRING ELSE NULL END) STORED, col1_14 STRING NOT NULL AS (CASE WHEN col1_6 IS NULL THEN e'%i$JNn\\n8':::STRING ELSE '':::STRING END) VIRTUAL, col1_15 STRING AS (CASE WHEN col1_3 IS NULL THEN 'Zm\"5>5o':::STRING ELSE e'\\x1b66EU]':::STRING END) VIRTUAL, \"co l1_16\" STRING NOT NULL AS (lower(CAST(col1_4 AS STRING))) VIRTUAL, PRIMARY KEY (col1_11 DESC, col1_2, \"col 1_12\" DESC), INVERTED INDEX (col1_6, col1_15 gin_trgm_ops ASC) WHERE ((((((((((ta͙ble_1.\"\\\\x49col1_8\" >= (-1):::INT8) AND (ta͙ble_1.col1_10 > 9223372036854775807:::INT8)) OR (ta͙ble_1.col1_13 != e'\\x00':::STRING)) OR (ta͙ble_1.col1_9 = '24:00:00-15:59':::TIMETZ)) AND (ta͙ble_1.col1_14 = 'X':::STRING)) AND (ta͙ble_1.\"col 1_12\" = (-1):::DECIMAL)) OR (ta͙ble_1.col1_11 > '\"':::STRING)) AND (ta͙ble_1.\"co l1_16\" = '':::STRING)) OR (ta͙ble_1.col1_15 <= e'\\'':::STRING)) OR (ta͙ble_1.col1_6 <= '4714-11-24 00:00:00 BC':::TIMESTAMP)) OR (ta͙ble_1.col1_2 > (-1.234E+401):::DECIMAL), UNIQUE (col1_3 DESC, (col1_2 + 8.991053112577277576:::DECIMAL) ASC) STORING (\"%vcol1_1\", col1_4, col1_6, \"\\\\x49col1_8\", col1_13), UNIQUE (\"co l1_16\") STORING (\"\\\\xa7c ol1_0\", \"%vcol1_1\", col1_4, col1_7) WHERE ((((((((((ta͙ble_1.col1_11 >= '':::STRING) OR (ta͙ble_1.col1_13 > 'X':::STRING)) OR (ta͙ble_1.\"co l1_16\" >= '\"':::STRING)) AND (ta͙ble_1.col1_9 = '24:00:00-15:59':::TIMETZ)) OR (ta͙ble_1.col1_14 > e'\\'':::STRING)) AND (ta͙ble_1.col1_6 <= '0001-01-01 00:00:00':::TIMESTAMP)) AND (ta͙ble_1.\"col 1_12\" = 'NaN':::DECIMAL)) OR (ta͙ble_1.\"\\\\x49col1_8\" >= (-32768):::INT8)) AND (ta͙ble_1.col1_15 != e'\\'':::STRING)) AND (ta͙ble_1.col1_10 != 32767:::INT8)) OR (ta͙ble_1.col1_2 != 'NaN':::DECIMAL), INVERTED INDEX (\"%vcol1_1\"), INDEX (col1_6) PARTITION BY LIST (col1_6) (PARTITION ta͙ble_1_part_0 VALUES IN (('1980-11-01 19:42:37.000361':::TIMESTAMP,), ('2026-03-07 19:47:21.000072':::TIMESTAMP,), ('2006-02-28 17:56:34.000741':::TIMESTAMP,), ('2007-02-15 03:07:27.000362':::TIMESTAMP,), ('1979-01-20 04:40:34.00054':::TIMESTAMP,), ('2009-03-23 01:03:07.000819':::TIMESTAMP,), ('1990-08-28 13:36:53.000842':::TIMESTAMP,), ('2002-11-19 09:42:48.000522':::TIMESTAMP,)), PARTITION ta͙ble_1_part_1 VALUES IN (('2019-08-04 11:11:28.000454':::TIMESTAMP,), ('1997-06-28 03:16:38.000211':::TIMESTAMP,), ('2003-10-15 07:16:46.000909':::TIMESTAMP,), ('2028-02-22 05:15:21.000476':::TIMESTAMP,), ('1976-12-23 01:42:35.000366':::TIMESTAMP,), ('2008-04-13 01:18:31.000007':::TIMESTAMP,), ('1997-08-05 01:27:50.000435':::TIMESTAMP,), ('1991-05-15 19:33:30.000567':::TIMESTAMP,)), PARTITION ta͙ble_1_part_2 VALUES IN (('1971-03-14 07:05:27.000288':::TIMESTAMP,), ('294276-12-31 23:59:59.999999':::TIMESTAMP,), ('1983-02-20 09:21:47.000377':::TIMESTAMP,), ('1980-09-13 01:05:34.000123':::TIMESTAMP,), ('1981-02-25 05:14:28.000367':::TIMESTAMP,), ('2001-08-19 08:36:51.000031':::TIMESTAMP,), ('2011-08-03 00:47:17.000694':::TIMESTAMP,), ('2003-11-02 10:22:04.000653':::TIMESTAMP,)), PARTITION \"ta͙bl?e_1😸_part_3\" VALUES IN (('2031-10-14 20:22:46.000038':::TIMESTAMP,), ('1974-02-10 18:36:20.000613':::TIMESTAMP,), ('2011-05-17 19:37:46.000443':::TIMESTAMP,), ('2027-01-01 03:27:48.000654':::TIMESTAMP,), ('2010-05-03 14:08:29.000191':::TIMESTAMP,), ('2023-10-29 15:23:01.000374':::TIMESTAMP,), ('1977-04-07 18:01:45.000137':::TIMESTAMP,), ('1988-08-21 07:21:34.000031':::TIMESTAMP,)), PARTITION ta͙ble_1_part_4 VALUES IN (('1994-01-25 05:52:28.00041':::TIMESTAMP,), ('1999-10-03 20:11:02.000418':::TIMESTAMP,), ('2000-01-14 17:37:54.000823':::TIMESTAMP,), ('2022-07-31 06:09:19.000592':::TIMESTAMP,), ('1985-01-08 02:44:56.000113':::TIMESTAMP,), ('1975-08-19 05:01:38.000605':::TIMESTAMP,), ('1998-11-12 07:25:10.000938':::TIMESTAMP,), ('2023-01-05 03:40:29.000585':::TIMESTAMP,)), PARTITION ta͙ble_1_part_5 VALUES IN (('2023-05-24 09:04:14.000397':::TIMESTAMP,), ('1970-05-17 18:20:45.00045':::TIMESTAMP,), ('1979-06-20 10:22:12.000247':::TIMESTAMP,), ('1998-07-14 06:31:26.000615':::TIMESTAMP,), ('2011-02-19 10:01:27.000682':::TIMESTAMP,), ('1984-09-13 03:27:14.000702':::TIMESTAMP,), ('1986-06-04 17:10:09.000177':::TIMESTAMP,), ('1990-03-12 20:55:31.000555':::TIMESTAMP,)), PARTITION \"ta͙ble_ 1_part_6\" VALUES IN (('2005-08-25 12:01:46.000951':::TIMESTAMP,), ('1986-12-29 09:27:42.000495':::TIMESTAMP,), ('2010-09-01 11:11:52.000222':::TIMESTAMP,), ('2030-10-01 13:30:49.000406':::TIMESTAMP,), ('2016-06-05 20:51:36.000279':::TIMESTAMP,), ('1991-08-26 11:24:21.00093':::TIMESTAMP,), ('1983-07-10 06:40:27.000855':::TIMESTAMP,), ('2018-08-18 13:04:31.000167':::TIMESTAMP,)), PARTITION \"DEFAULT\" VALUES IN ((DEFAULT,))) WHERE ((((((((((ta͙ble_1.col1_9 >= '00:00:00+15:59':::TIMETZ) AND (ta͙ble_1.col1_11 <= '':::STRING)) OR (ta͙ble_1.col1_13 < '':::STRING)) AND (ta͙ble_1.col1_10 < (-9223372036854775807):::INT8)) OR (ta͙ble_1.col1_14 > e'\\x00':::STRING)) AND (ta͙ble_1.col1_15 > e'\\'':::STRING)) AND (ta͙ble_1.col1_6 > '4714-11-24 00:00:00 BC':::TIMESTAMP)) OR (ta͙ble_1.col1_2 <= 1:::DECIMAL)) AND (ta͙ble_1.\"\\\\x49col1_8\" != 0:::INT8)) AND (ta͙ble_1.\"co l1_16\" <= '':::STRING)) AND (ta͙ble_1.\"col 1_12\" != 'NaN':::DECIMAL), FAMILY (col1_6, col1_4, col1_13), FAMILY (col1_9, col1_2, col1_3, \"\\\\xa7c ol1_0\"), FAMILY (co̞😥l1_5), FAMILY (col1_7), FAMILY (\"\\\\x49col1_8\"), FAMILY (\"%vcol1_1\"));\nCREATE TABLE table_2 (col2_0 TIME NOT NULL, \"col\\\\U00078AC32_1\" REGROLE, \"c%pol2_2\" BYTES NOT NULL, \"\\\\u76E7😛col2_3\" GEOGRAPHY NOT NULL, \"\\\\x36coͥl2_4\" BOX2D NOT NULL, col2_5 BIT(26) NOT NULL, col2_6 REGROLE, \" col2_7\" TIME NOT NULL, \"c%qol2_8\" FLOAT8, col2_9 VARCHAR NOT NULL, \"c!ol,2_10\" REGROLE NULL, \"%v\ncol2_11\" INT2 NULL, \"col2\\\\x89_12\" VARCHAR NOT NULL, \"col\r2_13\" TIMETZ NOT NULL, \"c\"\"ol2?_14\" TIME, colͦ2_15 STRING NOT NULL AS (CASE WHEN \"c%pol2_2\" IS NULL THEN e'\\x119':::STRING ELSE 'O':::STRING END) STORED, col2_16 STRING NOT NULL AS (lower(CAST(\"\\\\x36coͥl2_4\" AS STRING))) VIRTUAL, col2_17 STRING NOT NULL AS (lower(CAST(\"\\\\x36coͥl2_4\" AS STRING))) VIRTUAL, col2_18 STRING AS (CASE WHEN \"col\\\\U00078AC32_1\" IS NULL THEN '\"':::STRING ELSE e',7-|``9\\t':::STRING END) STORED, UNIQUE (\" col2_7\" DESC), FAMILY (colͦ2_15), FAMILY (\"c\"\"ol2?_14\"), FAMILY (col2_9, col2_6), FAMILY (\"col2\\\\x89_12\"), FAMILY (\"col\\\\U00078AC32_1\", \"\\\\x36coͥl2_4\", col2_5, \" col2_7\", \"col\r2_13\", \"c!ol,2_10\", \"%v\ncol2_11\", col2_0), FAMILY (\"c%qol2_8\", \"\\\\u76E7😛col2_3\", col2_18), FAMILY (\"c%pol2_2\"), UNIQUE (col2_9, \"col2\\\\x89_12\", colͦ2_15));\nCREATE TABLE table_3 (col3_0 TIMESTAMPTZ NOT NULL, \"c%pol3_1\" OID NULL, \"co&l3_2\" BIT(29) NOT NULL, \" col3_3\" OID NULL, col3_4 REFCURSOR NULL, col3_5 UUID NULL, 🙂col3_6 INT8 NOT NULL, col3_7 TIMETZ NULL, \"col3\"\"_8\" GEOGRAPHY[], col3_9 INT4 NOT NULL, col3_10 BOOL NOT NULL, col3_11 INT8 NOT NULL AS (🙂col3_6 + col3_9) VIRTUAL, INDEX (\"c%pol3_1\" DESC) STORING (col3_0, \"co&l3_2\", col3_4, col3_5, col3_7, \"col3\"\"_8\"), INDEX (lower(CAST(\" col3_3\" AS STRING)) DESC) WHERE (((((table_3.🙂col3_6 = 1:::INT8) OR (table_3.col3_0 != '0001-01-01 00:00:00+00':::TIMESTAMPTZ)) OR (table_3.col3_7 <= '00:00:00+15:59':::TIMETZ)) OR (table_3.col3_9 <= 1:::INT8)) OR (table_3.col3_11 >= (-32768):::INT8)) AND table_3.col3_10, FAMILY (col3_5), FAMILY (\" col3_3\"), FAMILY (\"c%pol3_1\"), FAMILY (col3_7, col3_10), FAMILY (col3_0), FAMILY (col3_4), FAMILY (🙂col3_6, \"co&l3_2\"), FAMILY (\"col3\"\"_8\"), FAMILY (col3_9));\nCREATE TABLE \"'table_4\" (\"&col4_0\" INET NOT NULL, \"c̼o*l4_1\" \"char\", PRIMARY KEY (\"&col4_0\" DESC), INVERTED INDEX (\"c̼o*l4_1\" gin_trgm_ops ASC) WHERE \"'table_4\".\"c̼o*l4_1\" = 'X':::STRING, INDEX (\"c̼o*l4_1\", \"&col4_0\" ASC) WHERE \"'table_4\".\"c̼o*l4_1\" = e'\\'':::STRING NOT VISIBLE, UNIQUE (\"&col4_0\" ASC), UNIQUE (\"&col4_0\" DESC) WHERE \"'table_4\".\"c̼o*l4_1\" != '':::STRING, FAMILY (\"&col4_0\"), FAMILY (\"c̼o*l4_1\"));\nCREATE TABLE \"t%vable_5\" (col̥5_0 DECIMAL NULL, \"col5 _1\" BOX2D NOT NULL, co🙀l5_2 DECIMAL NOT NULL, col5_3 INTERVAL NULL, col5_4 REGROLE, col5_5 VARCHAR, col5_6 BYTES NOT NULL, \"col5\"\"_7\" PG_LSN NOT NULL, col5_8 DECIMAL NULL AS (co🙀l5_2 + col̥5_0) VIRTUAL, PRIMARY KEY (\"col5\"\"_7\" ASC), UNIQUE (col5_3, col̥5_0 ASC) STORING (col5_5, col5_6) WHERE ((((\"t%vable_5\".col̥5_0 >= 1.0:::DECIMAL) OR (\"t%vable_5\".col5_8 < 1.0:::DECIMAL)) OR (\"t%vable_5\".col5_6 < '\\x27':::BYTES)) OR (\"t%vable_5\".co🙀l5_2 = (-1.0):::DECIMAL)) AND (\"t%vable_5\".col5_5 <= e'\\x00':::STRING));\nCREATE TABLE \"taBle_6\" (\"col6 _0\" DATE, \"co\vl6_1\" CHAR NOT NULL, col6_2 FLOAT4 NOT NULL, \"cOl6{_3\" REGTYPE NOT NULL, col6_4 REGCLASS NULL, col6_5 FLOAT4 NULL, \"\\\\u0F80col6_6\" FLOAT8 NOT NULL AS (col6_2 + (-0.2348167896270752):::FLOAT8) VIRTUAL, col6_7 FLOAT8 NULL AS (col6_5 + col6_2) STORED, col6_8 STRING AS (CASE WHEN \"cOl6{_3\" IS NULL THEN e'\\x00':::STRING ELSE 'O':::STRING END) STORED, c😈ol6_9 FLOAT8 NULL AS (col6_2 + col6_5) VIRTUAL, PRIMARY KEY (\"cOl6{_3\"), INDEX (\"\\\\u0F80col6_6\" ASC, \"cOl6{_3\", col6_7 DESC, col6_8 DESC, \"col6 _0\" DESC) WHERE (\"taBle_6\".col6_2 > 3.4028234663852886e+38:::FLOAT8) AND (\"taBle_6\".col6_5 > 'NaN':::FLOAT8), INDEX (col6_2 ASC) WHERE (((\"taBle_6\".col6_2 <= 0.0:::FLOAT8) OR (\"taBle_6\".col6_5 != (-1.0):::FLOAT8)) OR (\"taBle_6\".c😈ol6_9 = 5e-324:::FLOAT8)) OR (\"taBle_6\".col6_8 <= e'\\'':::STRING), INDEX (col6_4 DESC, c😈ol6_9, \"co\vl6_1\") WHERE (((((((\"taBle_6\".col6_2 <= 1.401298464324817e-45:::FLOAT8) AND (\"taBle_6\".\"co\vl6_1\" != e'\\x00':::STRING)) OR (\"taBle_6\".col6_5 > '+Inf':::FLOAT8)) OR (\"taBle_6\".col6_8 != e'\\x00':::STRING)) AND (\"taBle_6\".\"\\\\u0F80col6_6\" > 1.401298464324817e-45:::FLOAT8)) AND (\"taBle_6\".col6_7 <= 'NaN':::FLOAT8)) OR (\"taBle_6\".c😈ol6_9 < 1.7976931348623157e+308:::FLOAT8)) AND (\"taBle_6\".\"col6 _0\" = '4714-11-24 BC':::DATE), UNIQUE (col6_2 ASC) WHERE (((((((\"taBle_6\".c😈ol6_9 != '+Inf':::FLOAT8) OR (\"taBle_6\".\"co\vl6_1\" >= e'\\U00002603':::STRING)) OR (\"taBle_6\".col6_8 < e'\\x00':::STRING)) OR (\"taBle_6\".\"col6 _0\" < '4714-11-24 BC':::DATE)) OR (\"taBle_6\".col6_7 = 5e-324:::FLOAT8)) AND (\"taBle_6\".\"\\\\u0F80col6_6\" = 1.0:::FLOAT8)) OR (\"taBle_6\".col6_5 <= '+Inf':::FLOAT8)) AND (\"taBle_6\".col6_2 <= 3.4028234663852886e+38:::FLOAT8), FAMILY (\"co\vl6_1\"), FAMILY (col6_2, col6_8), FAMILY (\"col6 _0\", col6_7), FAMILY (\"cOl6{_3\"), FAMILY (col6_5, col6_4));\nCREATE TABLE 🙃table_7 (col7_0 INT4, \"col7\"\"_1\" BIT(37) NOT NULL, col7_2 REGTYPE, col7_3 TSVECTOR NOT NULL, col7_4 STRING COLLATE da_DK NOT NULL, \"col7\f_5\" INET NOT NULL, \"col\\v7_6\" STRING NOT NULL, col7_7 REFCURSOR NOT NULL, col͈7_8 BYTES NOT NULL, \"-col7_9\" TSVECTOR, \"\\\\xfdco̼l7_10\" TIMESTAMP NOT NULL, col7_11 BYTES NOT NULL, \"coL7_12\" REGCLASS, col7_13 OID NOT NULL, col7_14 INT4 NOT NULL, \"c\"\"ol7_15\" STRING NOT NULL, \"c%vol7_16\" INT8 NOT NULL AS (col7_0 + 1020007332:::INT8) VIRTUAL, PRIMARY KEY (\"col\\v7_6\", col7_13 ASC, \"\\\\xfdco̼l7_10\" DESC, \"c%vol7_16\" DESC), UNIQUE ((col7_14 + col7_0)) STORING (\"col7\"\"_1\", col7_3, \"col7\f_5\", col7_7, col͈7_8, \"coL7_12\", col7_14), UNIQUE (lower(CAST(col7_7 AS STRING)) ASC, \"col\\v7_6\" DESC, col7_4 DESC) STORING (\"col7\"\"_1\", \"col7\f_5\", \"-col7_9\", col7_11) WHERE (((((((🙃table_7.\"c%vol7_16\" >= (-1):::INT8) OR (🙃table_7.col͈7_8 < '\\x22':::BYTES)) AND (🙃table_7.col7_11 >= '\\x58':::BYTES)) OR (🙃table_7.col7_0 = 127:::INT8)) OR (🙃table_7.col7_14 = 2147483647:::INT8)) AND (🙃table_7.\"col\\v7_6\" = e'\\U00002603':::STRING)) OR (🙃table_7.\"c\"\"ol7_15\" = '\"':::STRING)) AND (🙃table_7.\"\\\\xfdco̼l7_10\" = '2001-01-01 00:00:00 BC':::TIMESTAMP), UNIQUE (\"col7\f_5\" DESC) STORING (col7_2, col7_3, col7_7, col͈7_8, col7_11, \"c\"\"ol7_15\"), UNIQUE (\"coL7_12\"), INDEX (col͈7_8 ASC), UNIQUE (\"col\\v7_6\" ASC) PARTITION BY LIST (\"col\\v7_6\") (PARTITION \"🙃%qtable_7_part_0\" VALUES IN ((e'\\U00002603':::STRING,), (e'+u1\\x11\\x14+\\r':::STRING,), ('':::STRING,), (e'6\\x1ebar\\x04g':::STRING,)), PARTITION 🙃table_7_part_1 VALUES IN (('X':::STRING,), (e'|Yu\\'f ``\\x7f(':::STRING,), ('?':::STRING,), (e'?w\\x18':::STRING,)), PARTITION 🙃table_7_part_2 VALUES IN ((e'\\x15erZ\\x1c!':::STRING,), (e'\\b@b-':::STRING,), ('v':::STRING,)), PARTITION 🙃table_7_part_3 VALUES IN ((e'/9Z\\x04{\\x18:':::STRING,), (e's\\x1ew-\\x0fQ+\\\\A':::STRING,), (e'J\\x0b$.h':::STRING,), ('3O8XI6':::STRING,)), PARTITION 🙃table_😙7_part_4 VALUES IN (('x':::STRING,), (e'/\\x02\\x18xLB$k':::STRING,), (e'p\\x1e\\x13\\x1d\\x1ac4\\x0b``':::STRING,), (e'L\\t<#\\x04\\x06H':::STRING,)), PARTITION 🙃table_7_part_5 VALUES IN ((e'T\\x07\\x18\\x078,':::STRING,), (e'\\x01c\\x1emm9\\x10\\n.':::STRING,), (e'\\x0b)\\tV;':::STRING,), (e';\\x17w':::STRING,)), PARTITION 🙃table_7_part🙁_6 VALUES IN ((e't\\x12':::STRING,), ('P':::STRING,)), PARTITION 🙃table_7_part_7 VALUES IN ((e'HC0\\x19P3\\x1eC':::STRING,), (e'\\x13_eGM':::STRING,), (e'hzAu\\x14\\x016MI':::STRING,), ('>TM6d?*F':::STRING,)), PARTITION 🙃table_7_part_8 VALUES IN (('IEG':::STRING,), (e'm\\x7f>5':::STRING,), (e'\\x07*\\x12;6Ty\\x02':::STRING,), ('zv#NM]':::STRING,)), PARTITION 🙃table_7_part_9 VALUES IN (('\"':::STRING,), ('wK':::STRING,))), UNIQUE (\"c%vol7_16\" DESC, col7_2 DESC) STORING (col7_3, \"col7\f_5\", col͈7_8, \"-col7_9\", col7_11, \"coL7_12\", col7_14) WHERE (((((((🙃table_7.col7_14 < (-32768):::INT8) OR (🙃table_7.col7_11 > '\\x22':::BYTES)) AND (🙃table_7.\"c%vol7_16\" >= (-9223372036854775807):::INT8)) AND (🙃table_7.\"col\\v7_6\" >= 'X':::STRING)) AND (🙃table_7.\"c\"\"ol7_15\" != '\"':::STRING)) AND (🙃table_7.col͈7_8 != '\\xff':::BYTES)) OR (🙃table_7.col7_0 = 127:::INT8)) OR (🙃table_7.\"\\\\xfdco̼l7_10\" = '4714-11-24 00:00:00 BC':::TIMESTAMP));\nCREATE TABLE \"t  able_8\" (\" col8_0\" INET NOT NULL, col8_1 BIT(6) NOT NULL, col8_2 DATE, \"c\"\"ol8'_3\" BOOL NOT NULL, c̭ol8_4 BYTES NOT NULL, \" col8_5\" TIME NOT NULL, col8_6 STRING COLLATE da_DK NULL, \"c?ol8_7\" JSONB, \"c ol8_8\" PG_LSN NOT NULL, \" col\t8 _9\" STRING NOT NULL AS (CASE WHEN c̭ol8_4 IS NULL THEN e'\\x7f\\x1a<\\x04':::STRING ELSE '-Zy)FN':::STRING END) STORED, PRIMARY KEY (\" col8_0\"), INDEX (col8_6) STORING (\" col8_5\", \"c?ol8_7\", \"c ol8_8\") WHERE (((\"t  able_8\".col8_2 = 'infinity':::DATE) AND (NOT \"t  able_8\".\"c\"\"ol8'_3\")) AND (\"t  able_8\".\" col8_5\" >= '00:00:00':::TIME)) AND (\"t  able_8\".\" col\t8 _9\" = e'\\U00002603':::STRING), INDEX (\"c ol8_8\", col8_6, \"c?ol8_7\" ASC, \" col8_5\" DESC, (CASE WHEN col8_2 IS NULL THEN 'XP:':::STRING ELSE e'.\\x07,X':::STRING END) ASC, c̭ol8_4 ASC, \" col8_0\" DESC, \" col\t8 _9\" DESC, col8_1 DESC, col8_2 ASC) PARTITION BY LIST (\"c ol8_8\", col8_6, \"c?ol8_7\", \" col8_5\") (PARTITION \"t  able_8_pa\"\"rt_0\" VALUES IN (('2C372685/33112193':::PG_LSN, e'\\U0008FF79\\U00021340\\U0001B802\\U000C7044\\U000AF94F\\U00000808\\U000481F6\\U000C4328' COLLATE da_DK, '[false, \"zWIg\"]':::JSONB, '06:05:02.752017':::TIME), ('2FEDECEB/F470AE43':::PG_LSN, e'\\U000CB1B3\\U0000363E\\U0009598F\\U000E1335\\U000200E6\\U00066F60\\U0003C25B' COLLATE da_DK, '{\"YO\": {\"5p7MowmJp\": [0.12763189596981978, false], \"X0\": null, \"a\": {\"b\": 0.09036542272054732, \"oDy\": \"a\"}, \"fRMf\": 0.06881944558261646, \"foo\": \"baz\", \"iQEGVWjr\": {\"8r6\": false, \"a\": true}}, \"c\": false, \"d50ryc\": {\"0KsWLLZn\": \"Tua8nlMU\", \"a\": {\"B0zDjfidT\": \"Pq5j\", \"kPU57A\": 0.4960538117524149}, \"csOj\": 1.3232106372288006}, \"oYeLzrNr\": \"Yi1ZtVw\"}':::JSONB, '15:38:40.149542':::TIME), ('760457D6/C429DFAD':::PG_LSN, e'\\U00015889\\U000D7C0E\\U000840A3\\U000672A0' COLLATE da_DK, '{}':::JSONB, '02:40:09.130329':::TIME)), PARTITION \"t  ͌able_8_part_1\" VALUES IN (('F4652A3D/EEE260C4':::PG_LSN, e'\\U00030070\\U000D77E2\\U000A84BE\\U000D7E3D\\U000671B2' COLLATE da_DK, '{\"LKqper\": false, \"QEWIFz\": [{\"foobar\": [0.8706283450116024], \"pY\": false}, null], \"bar\": [null], \"foobar\": {\"M1MuQimW\": [\"bar\", false], \"QRCr\": false, \"bar\": {\"9gJj\": 0.9013330094739844, \"WCMBkQV\": [1.455312661917666]}, \"baz\": null, \"foobar\": [[false], 3.377862230167758, null]}, \"mzujsJNTz\": {\"a\": \"tZ7MP\"}}':::JSONB, '13:43:15.847133':::TIME), ('0/0':::PG_LSN, e'\\U0000CB69' COLLATE da_DK, '{\"baz\": null, \"foobar\": false, \"vIYM\": true}':::JSONB, '07:29:14.472136':::TIME), ('47C775AA/9D66DCA9':::PG_LSN, e'\\U00008410\\U00065AAF\\U00021F40\\U00031761\\U0008CF3B\\U000851FD\\U00095D86' COLLATE da_DK, '{\"83dBqvX\": {\"ZEW\": \"Dpf\", \"foo\": \"hZIK0sZYP\"}, \"VF\": true, \"b\": null, \"jdu\": {\"960SXsJ\": null}}':::JSONB, '21:22:38.112763':::TIME)), PARTITION \"t  able_8_part_2\" VALUES IN (('40DAE85/6B20E90B':::PG_LSN, e'\\U000490A9\\U0001DA5C\\U00079763\\U0002A9E7\\U0001254C\\U0008BBCC\\U000D14A6' COLLATE da_DK, '[{\"0lTekDrp\": \"9p\", \"DwfMEGjo\": 0.9894423744004726, \"c\": \"c\"}, [{\"bar\": 0.7535567160055688}, \"MuGAJJ97\", true], true, true, \"b\"]':::JSONB, '14:30:36.550981':::TIME), ('83CB24AB/A6FDE214':::PG_LSN, e'\\U00016185' COLLATE da_DK, '[[2.115733942805724, true], true, null]':::JSONB, '16:48:08.734725':::TIME), ('FFFFFFFF/FFFFFFFF':::PG_LSN, e'\\U0004D249\\U000C9005\\U0005A53D\\U0008A346\\U000E40C0\\U0000BCD4\\U000D8BAD\\U0001A706' COLLATE da_DK, '{\"foobar\": null, \"vC\": null}':::JSONB, '00:00:00':::TIME))), UNIQUE (col8_2, col8_1 ASC) STORING (\"c\"\"ol8'_3\", c̭ol8_4, col8_6, \" col\t8 _9\") WHERE ((((\"t  able_8\".col8_2 = '1970-01-01':::DATE) AND \"t  able_8\".\"c\"\"ol8'_3\") AND (\"t  able_8\".c̭ol8_4 > '\\x':::BYTES)) OR (\"t  able_8\".\" col\t8 _9\" <= e'\\'':::STRING)) AND (\"t  able_8\".\" col8_5\" < '24:00:00':::TIME));\nCREATE TABLE table_9 (\"%17col9%_0\" BYTES NOT NULL, \"col9(_1\" FLOAT4 NULL, col9_2 JSONB NOT NULL, col9_3 OID, co😀l9_4 PG_LSN NOT NULL, \"_\\gc\rol9_5\" DATE NOT NULL, col9_6 STRING NOT NULL AS (lower(CAST(co😀l9_4 AS STRING))) STORED, col9_7 STRING NOT NULL AS (lower(CAST(col9_2 AS STRING))) VIRTUAL, \"co l9_8\" STRING NOT NULL AS (lower(CAST(col9_2 AS STRING))) VIRTUAL, col9_9 STRING AS (lower(CAST(col9_3 AS STRING))) VIRTUAL, \"co\\\\uD397l9_10\" STRING AS (lower(CAST(col9_3 AS STRING))) VIRTUAL, PRIMARY KEY (col9_2, col9_7 DESC, col9_6 ASC), FAMILY (\"_\\gc\rol9_5\"), FAMILY (col9_2, col9_3, \"%17col9%_0\"), FAMILY (\"col9(_1\", co😀l9_4, col9_6));\nCREATE TABLE \"ta\"\"ble _10\" (\" co\"\"l10_0\" REGCLASS NULL, col10_1 VARCHAR NOT NULL, col10_2 JSONB NOT NULL, col10_3 GEOGRAPHY NOT NULL, \"col10\"\"_4\" INET NOT NULL, \"co%ql10%v_5\" OID NULL, col10_6 DATE NOT NULL, c͍ol10_7 STRING, \"col1 0_8\" OID NULL, \"\"\"col10_9\" JSONB NOT NULL, \"col1 0_10\" TSQUERY NULL, \"c&ol10_11\" STRING NOT NULL, col10_12 JSONB NOT NULL, \"col%v10_13\" REGTYPE, col10_14 TIMESTAMPTZ NOT NULL, \"col\\n10_15\" STRING AS (CASE WHEN \"col%v10_13\" IS NULL THEN '-jwya':::STRING ELSE 'u4Qy':::STRING END) VIRTUAL, PRIMARY KEY (col10_12 DESC, \"\"\"col10_9\" ASC, \"c&ol10_11\" DESC), INDEX ((CASE WHEN \" co\"\"l10_0\" IS NULL THEN e'\\x1bgkv!\\x1a':::STRING ELSE ')':::STRING END)) WHERE (((((\"ta\"\"ble _10\".col10_6 >= 'infinity':::DATE) AND (\"ta\"\"ble _10\".col10_1 <= e'\\x00':::STRING)) OR (\"ta\"\"ble _10\".c͍ol10_7 > e'\\U00002603':::STRING)) OR (\"ta\"\"ble _10\".\"col\\n10_15\" >= '\"':::STRING)) OR (\"ta\"\"ble _10\".\"c&ol10_11\" = e'\\'':::STRING)) AND (\"ta\"\"ble _10\".col10_14 = '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ), INDEX (c͍ol10_7) PARTITION BY LIST (c͍ol10_7) (PARTITION \"ta\"\"ble _10_part_0\" VALUES IN ((e'\\x07':::STRING,), (e'I\\x110/Z\\x16':::STRING,), ('':::STRING,), (e'P\\x13Bfq':::STRING,), (e'\\r#Inj?!#\\x1c':::STRING,), ('I':::STRING,), ('FS':::STRING,), (e'\\U00002603':::STRING,)), PARTITION \"ta\"\"ble _\\g10_pa\tr t_1\" VALUES IN ((e'hD\\\\\\x06':::STRING,), (e'(zf*Bi\\x1e':::STRING,), ('Qs':::STRING,), (e'\\x15':::STRING,), (e'OU\\x0eJ\\x16':::STRING,), (e'uTeG\\b':::STRING,)), PARTITION \"ta{\"\"ble _10_part_2\" VALUES IN ((e'\\'':::STRING,), (e'w\\x0ev\\x06':::STRING,), ('?wr':::STRING,), (e'\\x00':::STRING,), ('8yY':::STRING,), (e'\\x15D\"\\x01UH':::STRING,), (e' H6a\\x1a':::STRING,), (e'\\x7fg':::STRING,)), PARTITION \"ta\"\"ble  _10_part_3\" VALUES IN (('sW~':::STRING,), (e'xHtd\\x1d\\x01':::STRING,), ('\"':::STRING,), (e'\\x023!/\\x14(_\\x05':::STRING,), (e'bH\\x19\\x15':::STRING,)), PARTITION \"ta\"\"ble _10'_part_4\" VALUES IN ((e'\\b\\x06\\x06 \\n\\b':::STRING,), (e'/|=0CW\\x04':::STRING,), (e'-\\'\\x16':::STRING,), (')':::STRING,), (e'KQ\\x17}\\x12+':::STRING,), (e'=c8[]C\\x0f$':::STRING,), (e'\\x1b:Nv':::STRING,), (e'k;\\n\\x1av|n!l':::STRING,)), PARTITION \"ta\"\"ble _10_part_5\" VALUES IN (('l&':::STRING,), ('%':::STRING,), ('XtS<':::STRING,), (e'[\\x176KVcl*':::STRING,), ('uY+':::STRING,), ('yQ':::STRING,), (e'\\x0fdr!':::STRING,), (e'@\\x1dKg\\x032\\x160':::STRING,)), PARTITION \"ta\"\"ble _10_part_6\" VALUES IN ((e'y\\x1a\\x18':::STRING,), (e'\\x1bS7\\x7f':::STRING,), (e'$l\\x11:G\\x10\\rR':::STRING,), (e'v*tC\\x16&':::STRING,), (e']RP\\x02':::STRING,), ('f':::STRING,), (e'>{\\x1f\\n=cl\\x1f':::STRING,), (e'_g8\\x0f!J':::STRING,)), PARTITION \"DEFAULT\" VALUES IN ((DEFAULT,))), INDEX (col10_1 DESC) STORING (c͍ol10_7, \"col1 0_8\", \"col1 0_10\", \"col%v10_13\"), UNIQUE (\"col%v10_13\") STORING (\" co\"\"l10_0\", col10_2, col10_3, \"col10\"\"_4\", \"co%ql10%v_5\", c͍ol10_7, \"col1 0_8\", \"col1 0_10\"));\nALTER TABLE table_9 ADD FOREIGN KEY (\"co l9_8\", \"co\\\\uD397l9_10\", col9_7) REFERENCES table_2 (col2_9, \"col2\\\\x89_12\", colͦ2_15);\n" args=[]: pq: unimplemented: virtual column "co l9_8" cannot reference a foreign key
    panic.go:626: -- test log scope end --
--- FAIL: TestPopulateTableWithRandData (3.19s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPopulateTableWithRandData.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/randgen/mutator.go
+++ b/pkg/sql/randgen/mutator.go
@@ -463,6 +463,12 @@ func foreignKeyMutator(
 		for _, def := range table.Defs {
 			switch def := def.(type) {
 			case *tree.ColumnTableDef:
+				if def.Computed.Virtual {
+					// We currently don't support FK references to / from
+					// virtual columns.
+					// TODO(#59671): remove this skip.
+					continue
+				}
 				idx, ok := tableNameToColIdx(table.Table)
 				if !ok {
 					idx = len(cols)

